coreax/coresubset.py

```diff
@@ -617,3 +615,296 @@ def _loop_body(
+            and dynamic values that are present in the class. The second is a dictionary
+            holding the static auxiliary data for the class, with keys being the names
+            of class attributes, and values being the values of the corresponding class
+            attributes.
+        """
+        children = (
+            self.random_key,
+            self.kernel,
+            self.stein_kernel,
+            self.coreset_indices,
+            self.coreset,
+        )
+        aux_data = {
+            "block_size": self.block_size,
+            "unique": self.unique,
+            "refine_method": self.refine_method,
+            "weights_optimiser": self.weights_optimiser,
+            "regularise": self.regularise,
+            "score_method": self.score_method,
+        }
+        return children, aux_data
+
+    def fit_to_size(self, coreset_size: int) -> None:
+        r"""
+        Execute Stein thinning algorithm with Jax.
+
+        We first compute a score function, and then the Stein kernel. This is used to
+        greedily choose points in the coreset to minimise kernel Stein discrepancy
+        (KSD).
+
+        :param coreset_size: The size of the of coreset to generate
+        """
+        # Check if coreset size is viable
+        if coreset_size <= 0 or isinstance(coreset_size, float):
+            raise ValueError("coreset_size must be a positive integer")
+        # Record the size of the original dataset
+        num_data_points = len(self.original_data.pre_coreset_array)
+
+        # Choose a score function
+        if self.score_method is None:
+            score_function = coreax.score_matching.KernelDensityMatching(
+                self.kernel.length_scale, self.original_data.pre_coreset_array
+            ).match(None)
+        else:
+            score_function = self.score_method.match(
+                self.original_data.pre_coreset_array
+            )
+
+        # Create a Stein kernel
+        if isinstance(self.kernel, coreax.kernel.SteinKernel):
+            self.stein_kernel = self.kernel
+            self.stein_kernel.score_function = score_function
+        else:
+            self.stein_kernel = coreax.kernel.SteinKernel(self.kernel, score_function)
+
+        # Initialise variables that will be updated throughout the loop. These are
+        # initially local variables, with the coreset indices being assigned to self
+        # when the entire set is created
+        kernel_similarity_penalty = jnp.zeros(num_data_points)
+        try:
+            # Note that a TypeError is raised if the size input to jnp.zeros is negative
+            coreset_indices = jnp.zeros(coreset_size, dtype=jnp.int32)
+
+        except TypeError as exception:
+            if coreset_size <= 0 or isinstance(coreset_size, float):
+                raise ValueError(
+                    "coreset_size must be a positive integer"
+                ) from exception
+            raise
+
```

Comment:
I think this try-except is now redundant due to the check on line 731

Reply:
Done.